src/lib/kwlib.sh

```diff
@@ -265,30 +265,23 @@ function get_kernel_version()
 function is_a_patch()
 {
   local -r FILE_PATH="$*"
-  local file_content
+  local cmd='git apply --check '
+  local cmd_return
 
   if [[ ! -f "$FILE_PATH" ]]; then
     return 1
   fi
 
-  file_content=$(< "$FILE_PATH")
+  cmd+=" ${FILE_PATH} > /dev/null 2>&1"
 
-  # The following array stores strings that are expected to be present
-  # in a patch file. The absence of any of these strings makes the
-  # given file be considered NOT a patch
-  local -ar PATCH_EXPECTED_STRINGS=(
-    'diff --git'
-    '---'
-    '@@'
-  )
+  cmd_manager 'SILENT' "$cmd"
+  cmd_return="$?"
 
-  for expected_str in "${PATCH_EXPECTED_STRINGS[@]}"; do
-    if [[ ! "$file_content" =~ $expected_str ]]; then
-      return 1
-    fi
-  done
+  if [[ "$cmd_return" == 0 || "$cmd_return" == 1 ]]; then
+    return 0
+  fi
```

Comment:
I'm confused here, 0 should be also considered? Also, update this function documentation to make it easier to follow why you return 0 and 1.